### PR TITLE
NPE when spouse is simply removed

### LINKED ISSUE #905: [0.44.0] NullPointerExceptions on Person select to view details (personnel log, etc).
### Issue
Receiving glitches and NullPointerExceptions on PersonView Selection in MekHQ ver. 0.44.0.

### Environment
MekHQ version: 0.44.0 (with custom files, also attached).
Operating System: Windows 10.
Java:
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

### Description
I'm trying to re-structure my LARGE force to a better force composition for me, but I am receiving some select glitches visually (picture attached) when selecting a few of my Mechwarrior pilots. I have similiar issues with other personnel outside of the Mechwarrior subtype, it is not local to the mechwarrior type. When requested to take a look at the log (also attached), I'm getting an NPE on PersonView.fillstats each time I select one that caused a glitch.

### Files
Megamek.0.44.0.Custom.Campagin.PersonSelectDetailsGraphicIssue.png
- PNG showing the visual glitch.
PhoenyxCullen_Issue.mm_0.44.0.zip
- Contains folder "PhoenyxCullen_Issue.mm_0.44.0" of which contains
  - Campaign subfolder with cpnx file in question.
  - data subfolder with custom mechs and universe data customs.

megamek 0 44 0 custom campagin personselectdetailsgraphicissue

[PhoenyxCullen_Issue.mm_0.44.0.zip](https://github.com/MegaMek/mekhq/files/2333348/PhoenyxCullen_Issue.mm_0.44.0.zip)

### PR DESCRIPTION
**Pull request by bandildo**
Fixes #905 

To check if some has spouse, use the new `Person.hasSpouse()` method, which checks if the actual spouse exists. Refrain from comparing `Person.getSpouse()` and `Person.getSpouseID()` with `null` as they are not safe. 

When a person is deleted and has a spouse, their spouse's spouseID will be set to null, just like in a divorce, to avoid this issue.

### CODE REVIEW

**Comment by AaronGullickson** [comment]
If `Person.getSpouse()` and `Person.getSpouseID()` are not safe, shouldn't we remove them? Also, best PR title ever.

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 1]
```diff
@@ -1004,10 +1005,18 @@ public void setSpouseID(UUID spouse) {
         this.spouse = spouse;
     }
 
+    @Nullable
     public Person getSpouse() {
         return campaign.getPerson(spouse);
     }
 
+    public boolean hasSpouse(){
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/personnel/Person.java]: A design where this is side effect free would be most desirable.

**Comment by bandildo** [comment]
@AaronGullickson they are used for multiple things, its just that using them as flag for checking if it has spouse is prone to error. Something like misusing the comparison a string to null, instead of using isEmpty.

**Comment by bandildo** [comment]
@HammerGS the changes requested by @sixlettervariables have already been addressed in the last commit. Hopefully all is good now.

**Pull request merged**